sdk/eventgrid/Azure.Messaging.EventGrid/src/MigrationGuide.md

```diff
@@ -0,0 +1,195 @@
+
+## Important changes
+
+### Package names and namespaces
+
+Package names and the namespace root for the modern Azure client libraries for .NET have changed. Each will follow the pattern Azure.[Area].[Service] where the legacy clients followed the pattern Microsoft.Azure.[Service]. This provides a quick and accessible means to help understand, at a glance, whether you are using the modern or legacy clients.
+
+In the case of Event Grid, the modern client libraries have packages and namespaces that begin with Azure.Messaging.EventGrid and were released beginning with version 4. The legacy client libraries have packages and namespaces that begin with Microsoft.Azure.EventGrid and a version of 3.x.x or below.
+
+### Client naming and constructors
+
+In Microsoft.Azure.EventGrid, the `EventGridClient` was instantiated by passing the `TopicCredentials` into the constructor:
+```C#
+string topicEndpoint = "https://<topic-name>.<region>-1.eventgrid.azure.net/api/events";
+string topicKey = "<topic-key>";
+
+TopicCredentials topicCredentials = new TopicCredentials(topicKey);
+EventGridClient client = new EventGridClient(topicCredentials);
+```
+
+In the `Azure.Messaging.EventGrid` library, the client used to interact with the service was renamed as the `EventGridPublisherClient` to highlight the fact that the client is only used to publish events and does not directly consume events from the service. The constructor takes the topic endpoint in addition to a credential parameter since a single client will only ever be able to publish to a single topic.
+```C# Snippet:CreateClientWithOptions
+EventGridPublisherClient client = new EventGridPublisherClient(
+    new Uri(topicEndpoint),
+    new AzureKeyCredential(topicAccessKey));
+```
+
+### Publishing events to a topic
+
+In the `Microsoft.Azure.EventGrid` library, events were published as shown:
+```C#
+string topicEndpoint = "https://<topic-name>.<region>-1.eventgrid.azure.net/api/events";
+string topicHostname = new Uri(topicEndpoint).Host;
+
+List<EventGridEvent> eventsList = new List<EventGridEvent>();
+    for (int i = 0; i < 1; i++)
+    {
+        eventsList.Add(new EventGridEvent()
+        {
+            Id = Guid.NewGuid().ToString(),
+            EventType = "Contoso.Items.ItemReceivedEvent",
+            Data = new ContosoItemReceivedEventData()
+            {
+                ItemUri = "ContosoSuperItemUri"
+            },
+
+            EventTime = DateTime.Now,
+            Subject = "Door1",
+            DataVersion = "2.0"
+        });
+    }
+
+await client.PublishEventsAsync(topicHostname, eventsList);
+```
+
+In `Azure.Messaging.EventGrid`, when publishing events it is no longer necessary to include the topic host name, as this is instead passed as part of the `Uri` that is provided to the `EventGridPublisherClient` constructor. The `EventGridEvent` type now has a constructor that includes parameters for all required fields of the event grid schema:
```

Comment:
```suggestion
In `Azure.Messaging.EventGrid`, when publishing events, it is no longer necessary to include the topic host name; this is instead passed as part of the `Uri` that is provided to the `EventGridPublisherClient` constructor. The `EventGridEvent` type now has a constructor that includes parameters for all required fields of the event grid schema:
```

Reply:
I think we can drop the semicolon and use a period instead.

Reply:
That's usually true, depending on how much you want to link the clauses.  Either way reads well enough to me.